### PR TITLE
chore: optimize get secret key from store

### PR DESCRIPTION
**Pull request by dongjiang1989**
## Description

_Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue._

optimize get secret key from store

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [x] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Verification

Please check the [Prometheus-Operator testing guidelines](../TESTING.md) for recommendations about automated tests.

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
optimize get secret key from store
```

### CODE REVIEW

File: pkg/prometheus/promcfg.go [hunk 1]
```diff
@@ -2525,7 +2525,7 @@ func (cg *ConfigGenerator) GenerateAgentConfiguration(
 }
 
 func (cg *ConfigGenerator) appendScrapeConfigs(
-	ctx context.Context,
+	_ context.Context,
```
[the remark this answers is not in the document]
Comment (dongjiang1989) [reply pkg/prometheus/promcfg.go:2528]: Yes. removed it done.

File: pkg/prometheus/promcfg.go [hunk 2]
```diff
@@ -2542,7 +2542,7 @@ func (cg *ConfigGenerator) appendScrapeConfigs(
 
 	for _, identifier := range scrapeConfigIdentifiers {
 		cfgGenerator := cg.WithKeyVals("scrapeconfig", identifier)
-		scrapeConfig, err := cfgGenerator.generateScrapeConfig(ctx, scrapeConfigs[identifier], store, shards)
+		scrapeConfig, err := cfgGenerator.generateScrapeConfig(scrapeConfigs[identifier], store, shards)
```
[the remark this answers is not in the document]
Comment (dongjiang1989) [reply pkg/prometheus/promcfg.go]: Thanks. Change *StoreBuilder to StoreGetter.
Done.

**Comment by dongjiang1989** [comment]
> Thanks for doing this!

Thanks. Please re-check it.  @simonpasquier

**Review by simonpasquier — APPROVED** [review]
Thanks!

**Pull request merged**